docs/architecture/adr-010-unified-cli-arguments-hermes.md

```diff
@@ -0,0 +1,173 @@
+# ADR 010: Unified approach for CLI arguments for Hermes v1.0.0
+
+## Changelog
+* 15.06.2022: Proposed.
+
+## Context
+
+In this ADR we provide recommendations and intuitions to using flags for all the arguments of the Hermes commands.
+
+The problem we are trying to solve is a unified approach to CLI arguments for Hermes v1.0.0.
+
+## Decision
+
+To avoid confusion, all the parameters should take long flags. The following general scenarios should be applied:
+
+* Only long flags are used in order to avoid having nonintuitive flags or conflicting flags.
+* Any parameter ending with `_id` should have the `_id` removed from the flag to shorten it. For example the flag for `chain_id` should only be `chain`.
+* Flags which can be shorten and still be meaninful should be shorten. This is done for `connection`, `channel` and `sequence`, which become respectively `conn`, `chan` and `seq`.
+* In cases where there are source and destination parameters for a same object, the flags start with the prefix `--src-` and `--dst-`.
+* In cases where there are undirectional parameters for a same object, the flags end with the suffix `-a` and `-b`.
```

Comment:
My choice of word is confusing sorry. My interpretation of nondirectional parameters are the ones where using parameters `--a A` and `--b B` will result the same as `--a B` and `--b A`.

Let's use `create connection` as an example. There are two chains `ibc-0` and `ibc-1`. There will have be an internal difference between `--chain-a ibc-0` and `--chain-b ibc-1` and `--chain-a ibc-1` and `--chain-b ibc-0`, which is the chain that initialises the handshake. But the created connection is the same.

This is the reason why `src` and `dst` seemed confusing if used for all the cases.

Reply:
> My choice of word is confusing sorry. My interpretation of nondirectional parameters are the ones where using parameters --a A and --b B will result the same as --a B and --b A.

But the result is the same. I propose we do not use unidirectional. We can say that A and B identify entities (chains, clients, connections, ports) on the two ends of a connection or channel. We can just mention that `OpenInit` is sent to chain A.

Reply:
> Also we have `src_chain` but `chain_a`. In the code we use `src_chain` and `a_chain`. The qualifier should be either prefix or suffix but not both.

I'm a bit confused by this as there is a mix of both `chain_a` and `a_chain` in the code. From a quick search I only see `a_chain` in the `relayer` create. But `chain_a` is in both the `relayer` and `relayer-cli` crates. If only prefix or suffix should be used, I find the prefix better, mostly for source and destination.

> > My choice of word is confusing sorry. My interpretation of nondirectional parameters are the ones where using parameters --a A and --b B will result the same as --a B and --b A.
> 
> But the result is the same. I propose we do not use unidirectional. We can say that A and B identify entities (chains, clients, connections, ports) on the two ends of a connection or channel. We can just mention that `OpenInit` is sent to chain A.

I will change my phrasing to: "In cases where there are two separate parameters for a same object type" would that seem better ?

Reply:
Maybe we should be very explicit. "For the channel and connection creation CLIs, the objects at the two ends are suffixed (or prefixed)  by `a` or `b`."

Reply:
Seems like a good idea. I'll change it to be explicit and at the same time change so that `a` and `b` are prefixes, to be coherent with the `host` and `reference`

Reply:
Yes `a_chain` sounds a bit strange in english. But I am bugged by the inconsistency. 
I know it's stretching it but one can also thing about `a` as an adjective...like in `first`, e.g. `first_chain` :)  
I have a slight preference for `<qualifier>_<object_name>` where the `<qualifier>` can be a tag or adjective.

All said, if you feel strongly about this I am ok with having it the way you propose.

Reply:
Perhaps a middle ground is to suffix everything? I'm personally okay with that:
```shell
$ hermes create client --chain-host ibc-1 --chain-reference ibc-2
$ hermes create connection --chain-a ibc-1 --chain-b ibc-2
```

Reply:
I put them to suffixes, let me know if this is preferred.

Reply:
i read the doc again and not super happy with how `chain-host` and `chain-reference` read. The latter for example reads "the reference of the chain". My order of preference now is:

1. `a-chain`, `host-chain`, `reference-chain`
2. `chain-a`, `host-chain`, `reference-chain` ...even though i think it's not consistent i now prefer it slightly over 3.
3. `chain-a`, `chain-host`, `chain-reference`

@adizere, @mircea-c et al. wdyt?

I will let you all decide, in the end I am ok with any of the above and will be much better than what we have today.

Reply:
I am partial to the first option. Only thing I don't like is the suffix for the `unidirectional`. Could just be `chain` without a suffix